vpr/src/route/rr_graph_reader.cpp

```diff
@@ -477,16 +477,17 @@ void verify_grid(pugi::xml_node parent, const pugiutil::loc_data & loc_data, con
 
         if (grid_tile.type->index != get_attribute(grid_node, "block_type_id", loc_data).as_float(0)) {
             vpr_throw(VPR_ERROR_OTHER, __FILE__, __LINE__,
-                    "Architecture file does not match RR graph's block_type_id");
+                    "Architecture file does not match RR graph's block_type_id at (%d, %d): arch used ID %d, RR graph used ID %d.", x, y,
+                     (grid_tile.type->index), (int)get_attribute(grid_node, "block_type_id", loc_data).as_float(0));
```

Comment:
I think we could we drop the `(int)` cast and just use `.as_int(0)`. Unless I'm missing something the `block_type_id` should always be an int.

Reply:
Yes, that makes sense. Would it be sensible to change it in the comparison too for sake of consistency (I agree, there's no sensible reason I can think of where it would be a float).

Reply:
Yep it would be good to swap both the comparison and print to use `.as_int()`.

With that it should be good to merge.

Reply:
Implemented in e666cccea